packages/at_client/lib/src/at_collection/at_collection_spec.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:at_client/at_client.dart';
+import 'package:at_client/src/at_collection/at_collection_impl.dart';
+import 'package:at_client/src/at_collection/model/at_operation_item_status.dart';
+
+abstract class AtCollectionSpec<T extends AtCollectionModel> {
+  /// Saves the object [T] in a self key.
+  /// A unique id is generated and stored in [T.id].
+  ///
+  /// Returns true if operation was successful else returns false.
+  ///
+  /// [expiryTime] Represents the time in milliseconds beyond which the key expires.
+  Future<bool> save(T model, {int? expiryTime});
```

Comment:
Few questions:

1. Does the save fail? If yes, how do we report it?
    - My suggestion: Let us throw whatever exceptions that atClient.put throws and document the same here.
2. Should we expose ttb also on ATCollectioonModel (ttl, ttb)
   - My suggestion: Let us make ttl and ttb available as attributes/setters on ATCollectionModel
3. Should we make void a return type for save()?
 - Mu suggestion: Yes, as anything that is false is reported as AtException